### PR TITLE
fix(masthead/footer/dotcom-shell): storybook examples and update components

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#5045 

### Description

Align masthead, footer, and dotcom-shell storybook examples with react. 

### Changelog

**New**

- add properties to masthead so users can toggle on and off profile and search
- add properties to dotcom-shell to include multiple properties from masthead and footer
- add micro-footer with language selector story
- update masthead stories to include the same knobs as react
- update custom nav links to match react
- update dotcom-shell storybook with multiple variations of the masthead and footer
- update footer checks for micro footer and disable-locale-button

**Changed**

- update README with L1 info

### CODE REVIEW

File: packages/web-components/src/components/dotcom-shell/dotcom-shell-composite.ts [hunk 1]
```diff
@@ -103,6 +104,18 @@ class DDSDotcomShellComposite extends LitElement {
    */
   _setLanguage?: (language: string) => void;
 
+  /**
+   * `true` if there is a profile.
+   */
+  @property({ type: Boolean, attribute: 'profile' })
+  profile = true;
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/dotcom-shell-composite.ts:111]:
```suggestion
  @property({ type: Boolean, attribute: 'has-profile' })
  hasProfile = true;
```

File: packages/web-components/src/components/dotcom-shell/dotcom-shell-composite.ts [hunk 2]
```diff
@@ -103,6 +104,18 @@ class DDSDotcomShellComposite extends LitElement {
...
+
+  /**
+   * `true` if there is a search.
+   */
+  @property({ type: Boolean, attribute: 'search' })
+  search = true;
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/dotcom-shell-composite.ts:117]:
```suggestion
  @property({ type: Boolean, attribute: 'has-search' })
  hasSearch = true;
```

File: packages/web-components/src/components/masthead/masthead-composite.ts [hunk 3]
```diff
@@ -471,6 +471,18 @@ class DDSMastheadComposite extends LitElement {
    */
   _setLanguage?: (language: string) => void;
 
+  /**
+   * `true` if there is a profile.
+   */
+  @property({ type: Boolean, attribute: 'profile' })
+  profile = true;
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/masthead-composite.ts:478]:
```suggestion
  @property({ type: Boolean, attribute: 'has-profile' })
  hasProfile = true;
```

File: packages/web-components/src/components/masthead/masthead-composite.ts [hunk 4]
```diff
@@ -471,6 +471,18 @@ class DDSMastheadComposite extends LitElement {
...
+
+  /**
+   * `true` if there is a search.
+   */
+  @property({ type: Boolean, attribute: 'search' })
+  search = true;
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/masthead-composite.ts:484]:
```suggestion
  @property({ type: Boolean, attribute: 'has-search' })
  hasSearch = true;
```

File: packages/web-components/src/components/masthead/masthead-composite.ts [hunk 5]
```diff
@@ -486,7 +498,7 @@ class DDSMastheadComposite extends LitElement {
   /**
    * The platform name.
    */
-  @property()
+  @property({ attribute: 'platform' })
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/masthead-composite.ts:501]:
```suggestion
  @property()
```

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 6]
```diff
@@ -61,31 +78,588 @@ export const Default = ({ parameters }) => {
...
+            user-status="${ifNonNull(userStatus)}"
+            searchPlaceHolder="${searchPlaceholder}"
+            selected-menu-item="${selectedMenuItem}"
+            .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
+            .legalLinks="${ifNonNull(legalLinks)}"
+            .localeList="${ifNonNull(localeList)}"
+            .footerLinks="${ifNonNull(footerLinks)}"
+            .navLinks="${navLinks}"
+            .profile="${profile}"
+            .search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:370]: Should use boolean attribute binding.

**Review by asudoh — COMMENTED** [review]
Nice work - Thanks @ariellalgilmore!

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts (hunk 6 above)
Comment (ariellalgilmore) [reply packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts]: updated!

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 7]
```diff
@@ -61,31 +78,588 @@ export const Default = ({ parameters }) => {
...
             user-status="${ifNonNull(userStatus)}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
+            selected-menu-item="${ifNonNull(selectedMenuItem)}"
             .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
             .legalLinks="${ifNonNull(legalLinks)}"
             .localeList="${ifNonNull(localeList)}"
             .footerLinks="${ifNonNull(footerLinks)}"
             .navLinks="${navLinks}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:94]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 8]
```diff
@@ -61,31 +78,588 @@ export const Default = ({ parameters }) => {
...
             .unauthenticatedProfileItems="${ifNonNull(unauthenticatedProfileItems)}"
+            ?disable-locale-button="${localeButton}"
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:96]:
```suggestion
            ?disable-locale-button="${disableLocaleButton}"
```

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 9]
```diff
@@ -61,31 +78,588 @@ export const Default = ({ parameters }) => {
...
+          </dds-dotcom-shell-container>
+        `}
+  `;
+};
+
+Default.story = {
+  parameters: {
+    knobs: {
+      FooterComposite: ({ groupId }) => ({
+        localeButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:129]:
```suggestion
        disableLocaleButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 10]
```diff
@@ -61,31 +78,588 @@ export const Default = ({ parameters }) => {
...
+        `}
+  `;
+};
+
+withPlatform.story = {
+  parameters: {
+    knobs: {
+      MastheadComposite: ({ groupId }) => ({
+        profile: boolean('show the profile functionality (profile)', true, groupId),
+        search: boolean('show the search functionality (search)', true, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:334]:
```suggestion
        hasProfile: boolean('show the profile functionality (profile)', true, groupId),
        hasSearch: boolean('show the search functionality (search)', true, groupId),
```

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 11]
```diff
@@ -61,31 +78,588 @@ export const Default = ({ parameters }) => {
...
+          </dds-dotcom-shell-container>
+        `}
+  `;
+};
+
+withMicroFooterLanguageOnly.story = {
+  parameters: {
+    knobs: {
+      FooterComposite: ({ groupId }) => ({
+        localeButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:605]:
```suggestion
        disableLocaleButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 12]
```diff
@@ -112,12 +686,17 @@ export default {
     ...readme.parameters,
     useRawContainer: true,
     knobs: {
+      escapeHTML: false,
       MastheadComposite: ({ groupId }) => ({
-        platform: textNullable('Platform (platform)', '', groupId),
-        logoHref: 'https://www.ibm.com',
+        platform: select('Platform (platform)', { none: null, platform: platformData.name }, null, groupId),
+        profile: boolean('show the profile functionality (profile)', true, groupId),
+        search: boolean('show the search functionality (search)', true, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:693]:
```suggestion
        hasProfile: boolean('show the profile functionality (profile)', true, groupId),
        hasSearch: boolean('show the search functionality (search)', true, groupId),
```

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 13]
```diff
@@ -112,12 +686,17 @@ export default {
...
+        searchPlaceholder: textNullable('search placeholder (searchPlaceholder)', 'Search all of IBM', groupId),
+        selectedMenuItem: textNullable('selected menu item (selected-menu-item)', 'Services & Consulting', groupId),
+        userStatus: select('The user authenticated status (user-status)', userStatuses, userStatuses.unauthenticated, groupId),
       }),
       FooterComposite: ({ groupId }) => ({
-        footerSize: select('Size (footer-size)', footerSizes, null, groupId),
+        localeButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:699]:
```suggestion
        disableLocaleButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```

File: packages/web-components/src/components/footer/__stories__/footer.stories.ts [hunk 14]
```diff
@@ -91,11 +97,21 @@ export const defaultLanguageOnly = ({ parameters }) => {
...
-    langList: mockLangList,
   };
   return base({ parameters });
 };
 
+defaultLanguageOnly.story = {
+  parameters: {
+    knobs: {
+      FooterComposite: ({ groupId }) => ({
+        localeButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/footer/__stories__/footer.stories.ts:108]:
```suggestion
        disableLocaleButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```

File: packages/web-components/src/components/footer/__stories__/footer.stories.ts [hunk 15]
```diff
@@ -106,15 +122,35 @@ export const short = ({ parameters }) => {
...
-    langList: mockLangList,
+  };
+  return base({ parameters });
+};
+
+shortLanguageOnly.story = {
+  parameters: {
+    knobs: {
+      FooterComposite: ({ groupId }) => ({
+        localeButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/footer/__stories__/footer.stories.ts:141]:
```suggestion
        disableLocaleButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```

File: packages/web-components/src/components/footer/__stories__/footer.stories.ts [hunk 16]
```diff
@@ -136,11 +175,27 @@ export const micro = ({ parameters }) => {
   `;
 };
 
+microLanguageOnly.story = {
+  parameters: {
+    knobs: {
+      FooterComposite: ({ groupId }) => ({
+        localeButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/footer/__stories__/footer.stories.ts:182]:
```suggestion
        disableLocaleButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```

File: packages/web-components/src/components/footer/__stories__/footer.stories.ts [hunk 17]
```diff
@@ -136,11 +175,27 @@ export const micro = ({ parameters }) => {
...
+};
+
 export default {
   title: 'Components/Footer',
   parameters: {
     ...readme.parameters,
     useRawContainer: true,
+    knobs: {
+      FooterComposite: ({ groupId }) => ({
+        localeButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/footer/__stories__/footer.stories.ts:196]:
```suggestion
        disableLocaleButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 18]
```diff
@@ -48,9 +49,11 @@ export const Default = ({ parameters }) => {
             platform-url="${ifNonNull(platformData.url)}"
             selected-menu-item="${ifNonNull(selectedMenuItem)}"
             user-status="${ifNonNull(userStatus)}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
             .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:55]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 19]
```diff
@@ -60,14 +63,18 @@ export const Default = ({ parameters }) => {
             platform-url="${ifNonNull(platformData.url)}"
             selected-menu-item="${ifNonNull(selectedMenuItem)}"
             user-status="${ifNonNull(userStatus)}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
             .navLinks="${navLinks}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:69]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 20]
```diff
@@ -77,15 +84,59 @@ export const WithCustomData = ({ parameters }) => {
       platform-url="${ifNonNull(platformData.url)}"
       selected-menu-item="${ifNonNull(selectedMenuItem)}"
       user-status="${ifNonNull(userStatus)}"
+      searchPlaceholder="${ifNonNull(searchPlaceholder)}"
       .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
       .navLinks="${customLinks}"
+      ?has-profile="${profile}"
+      ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:91]:
```suggestion
      ?has-profile="${hasProfile}"
      ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 21]
```diff
@@ -77,15 +84,59 @@ export const WithCustomData = ({ parameters }) => {
...
+            activate-search="true"
+            platform="${ifNonNull(platformData.name)}"
+            platform-url="${ifNonNull(platformData.url)}"
+            selected-menu-item="${ifNonNull(selectedMenuItem)}"
+            user-status="${ifNonNull(userStatus)}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
+            .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
+            .navLinks="${navLinks}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:117]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 22]
```diff
@@ -77,15 +84,59 @@ export const WithCustomData = ({ parameters }) => {
...
+          <dds-masthead-container
+            activate-search="true"
+            platform="${ifNonNull(platform)}"
+            platform-url="${ifNonNull(platformData.url)}"
+            selected-menu-item="${ifNonNull(selectedMenuItem)}"
+            user-status="${ifNonNull(userStatus)}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
+            .navLinks="${navLinks}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:131]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 23]
```diff
@@ -98,8 +149,11 @@ export const withPlatform = ({ parameters }) => {
             platform-url="${ifNonNull(platformData.url)}"
             selected-menu-item="${ifNonNull(selectedMenuItem)}"
             user-status="${ifNonNull(userStatus)}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
             .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
             .navLinks="${navLinks}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:156]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 24]
```diff
@@ -109,7 +163,10 @@ export const withPlatform = ({ parameters }) => {
             platform-url="${ifNonNull(platformData.url)}"
             selected-menu-item="${ifNonNull(selectedMenuItem)}"
             user-status="${ifNonNull(userStatus)}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
             .navLinks="${navLinks}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:169]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 25]
```diff
@@ -119,16 +176,19 @@ withPlatform.story = {
   parameters: {
     knobs: {
       MastheadComposite: ({ groupId }) => ({
+        profile: boolean('show the profile functionality (profile)', true, groupId),
+        search: boolean('show the search functionality (search)', true, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:180]:
```suggestion
        hasProfile: boolean('show the profile functionality (has-profile)', true, groupId),
        hasSearch: boolean('show the search functionality (has-search)', true, groupId),
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 26]
```diff
@@ -140,8 +200,11 @@ export const withL1 = ({ parameters }) => {
             platform="${ifNonNull(platform)}"
             platform-url="${ifNonNull(platformData.url)}"
             selected-menu-item="${ifNonNull(selectedMenuItem)}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
             user-status="${ifNonNull(userStatus)}"
             .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:207]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 27]
```diff
@@ -153,6 +216,8 @@ export const withL1 = ({ parameters }) => {
             platform-url="${ifNonNull(platformData.url)}"
             selected-menu-item="${ifNonNull(selectedMenuItem)}"
             user-status="${ifNonNull(userStatus)}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:220]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 28]
```diff
@@ -174,10 +240,12 @@ export const withAlternateLogoAndTooltip = ({ parameters }) => {
             platform-url="${ifNonNull(platformData.url)}"
             selected-menu-item="${ifNonNull(selectedMenuItem)}"
             user-status="${ifNonNull(userStatus)}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
             .authenticatedProfileItems="${ifNonNull(authenticatedProfileItems)}"
-            .l1Data="${l1Data}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:246]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 29]
```diff
@@ -187,14 +255,37 @@ export const withAlternateLogoAndTooltip = ({ parameters }) => {
             platform-url="${ifNonNull(platformData.url)}"
             selected-menu-item="${ifNonNull(selectedMenuItem)}"
             user-status="${ifNonNull(userStatus)}"
-            .l1Data="${l1Data}"
+            searchPlaceholder="${ifNonNull(searchPlaceholder)}"
             .navLinks="${navLinks}"
-            .logoData="${logoData}"
+            .logoData="${mastheadLogo}"
+            ?has-profile="${profile}"
+            ?has-search="${search}"
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:262]:
```suggestion
            ?has-profile="${hasProfile}"
            ?has-search="${hasSearch}"
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 30]
```diff
@@ -187,14 +255,37 @@ export const withAlternateLogoAndTooltip = ({ parameters }) => {
...
   `;
 };
 
+withAlternateLogoAndTooltip.story = {
+  parameters: {
+    knobs: {
+      MastheadComposite: ({ groupId }) => ({
+        platform: select('Platform (platform)', { none: null, platform: platformData.name }, null, groupId),
+        profile: boolean('show the profile functionality (profile)', true, groupId),
+        search: boolean('show the search functionality (search)', true, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:274]:
```suggestion
        hasProfile: boolean('show the profile functionality (has-profile)', true, groupId),
        hasSearch: boolean('show the search functionality (has-search)', true, groupId),
```

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 31]
```diff
@@ -214,11 +305,14 @@ export default {
     ...readme.parameters,
     'carbon-theme': { disabled: true },
     knobs: {
+      escapeHTML: false,
       MastheadComposite: ({ groupId }) => ({
         platform: select('Platform (platform)', { none: null, platform: platformData.name }, null, groupId),
+        profile: boolean('show the profile functionality (profile)', true, groupId),
+        search: boolean('show the search functionality (search)', true, groupId),
```
Comment (asudoh) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:312]:
```suggestion
        hasProfile: boolean('show the profile functionality (has-profile)', true, groupId),
        hasSearch: boolean('show the search functionality (has-search)', true, groupId),
```

**Review by asudoh — COMMENTED** [review]
Thanks for the updates @ariellalgilmore!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @ariellalgilmore for all the updates!

**Pull request merged**